budgetApp1/budgetApp1/LoginViewController.swift

```diff
@@ -1,45 +1,36 @@
-//
-//  LoginViewController.swift
-//  budgetApp1
-//
-//  Created by Lehlohonolo Nkadimeng on 2024/04/17.
-//
-
 import UIKit
 
 class LoginViewController: UIViewController {
-    @IBOutlet weak var usernameField: UITextField!
-    @IBOutlet weak var passwordField: UITextField!
     
-    var user1 = user.init(username: "", password: "", fullname: "", emailAddress: "")
+    // MARK: - IBOutlets
+    @IBOutlet private weak var usernameField: UITextField!
+    @IBOutlet private weak var passwordField: UITextField!
+    
+    let userVM = UserViewModel.shared // Use shared instance
```

Comment:
I am using a singleton because for some reason when I call the user stored in the userVM in the Login screen, it is empty. I think it is because when I instantiate the userVM again in the login after the user has signed in, it loses the value stored in User so I thought using a singleton would make the details persistent across both Login and SignUp Controllers. 
I know it's bad coding principles but I just had to find a way around this.

Reply:
lets have a call. I'm sure we can get it working the right way :)

Reply:
Okay

Reply:
had a chat with you on a call. We will use coredata in the next PR